packages/forklift-console-plugin/src/modules/Providers/views/details/components/DetailsSection/components/UIDetailsItem.tsx

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import { useModal } from 'src/modules/Providers/modals';
+import { EditProviderUIModal } from 'src/modules/Providers/modals/EditProviderUI';
+import { useForkliftTranslation } from 'src/utils/i18n';
+
+import { ExternalLink } from '@kubev2v/common';
+import { V1beta1Provider } from '@kubev2v/types';
+
+import { DetailsItem } from '../../../../../utils';
+
+import { ProviderDetailsItemProps } from './ProviderDetailsItem';
+
+/**
+ * @typedef {Object} UIDetailsItemProps - extends ProviderDetailsItemProps
+
+ * @property {string} [text] - A text to be displayed as a content.
+ * @property {function} [calcProviderUI] - A method for auto calculating the provider's UI path.
+ **/
+export interface UIDetailsItemProps extends ProviderDetailsItemProps {
+  text: string;
+  calcProviderUI: (provider: V1beta1Provider) => string;
+}
+
+const getProviderUIAnnotation = (provider: V1beta1Provider): string =>
+  provider?.metadata?.annotations?.['forklift.konveyor.io/providerUI'];
+
+export const UIDetailsItem: React.FC<UIDetailsItemProps> = ({
+  resource: provider,
+  canPatch,
+  text,
+  calcProviderUI,
```

Comment:
i like the concept of passing a method here, i have concern about the name `ProviderUI` sounds like some UI thing, we are doing a "web UI" everything here is a UI the components are UI componenet the application is a UI application, etc ... , we need to be clear that UI here is in the context of a link to a web based management system vs the context of some other UI thing we are doing.

Reply:
`calcProviderWebUIConsole` maybe? 
I didn't want to use long names, but we can think of something

Reply:
I prefer somthing with "link" in it, so we know it's a URL link string for the console and not the "console", any idea ?